tests/components/backup/test_init.py

```diff
@@ -14,12 +15,19 @@ async def test_setup_with_hassio(
     caplog: pytest.LogCaptureFixture,
 ) -> None:
     """Test the setup of the integration with hassio enabled."""
-    assert not await setup_backup_integration(hass=hass, with_hassio=True)
+    assert await setup_backup_integration(
+        hass=hass,
+        with_hassio=True,
+        configuration={DOMAIN: {}},
+    )
     assert (
         "The backup integration is not supported on this installation method, please"
         " remove it from your configuration"
     ) in caplog.text
 
+    assert DOMAIN in hass.data
+    assert isinstance(hass.data[DOMAIN], BackupManager)
```

Comment:
I'd remove these asserts. It's enough to test that we can do all the things that we should be able to do when `with_hassio` is True, which is tested in the websocket tests.